src/ralph/cli.py

```diff
@@ -177,6 +180,83 @@ def wrapper(command):
     return wrapper
 
 
+@cli.command()
+@click.option(
+    "-u",
+    "--username",
+    type=str,
+    required=True,
+    help="The user for which we generate credentials.",
+)
+@click.password_option(
+    "-p",
+    "--password",
+    type=str,
+    required=True,
+    help="The password to encrypt for this user. Will be prompted if missing.",
+)
+@click.option(
+    "-s",
+    "--scope",
+    type=str,
+    required=True,
+    multiple=True,
+    default=[],
+    help="The user scope(s). This option can be provided multiple times.",
+)
+@click.option(
+    "-w",
+    "--write",
+    is_flag=True,
+    default=False,
+    help="Write new credentials to the LRS authentication file.",
+)
+def auth(username, password, scope, write):
+    """Generate credentials for LRS HTTP basic authentification."""
+    logger.info("Will generate credentials for user: %s", username)
+
+    credentials = {
+        "username": username,
+        "hash": bcrypt.hashpw(
+            bytes(password, encoding="utf-8"), bcrypt.gensalt()
+        ).decode("ascii"),
+        "scopes": scope,
+    }
+
+    if write:
+        logger.info("Will append new credentials to: %s", settings.AUTH_FILE)
+
+        # Force Path object instantiation so that the file creation can be
+        # faked in a test environment.
+        auth_file = Path(settings.AUTH_FILE)
+        # Create the authentication file if it does not exist
+        auth_file.touch()
+        auth_file_content = auth_file.read_text(encoding="utf-8")
+
+        users = []
+        if len(auth_file_content):
+            try:
+                users = json.loads(auth_file_content)
+            except json.JSONDecodeError as exc:
+                logger.error(
+                    "Authentication file <%s> contains invalid JSON data!",
+                    settings.AUTH_FILE,
+                )
+                raise exc
+
+        users.append(credentials)
```

Comment:
```suggestion
        users.insert(0, credentials)
```

By inserting the new user at the top of the list we let the user to redefine his password (as the LRS only takes the first matching username occurrence for authentication). WDYT?

Reply:
That's seems convoluted and not straightforward. What about validating the credentials file instead? (_i.e._ ensuring that the username field value is unique).

Reply:
I agree) With validation, it would be much better)

Reply:
I've added a fixup commit that implements Pydantic models for credentials file validation. WDYT?